test/core/end2end/generate_tests.bzl

```diff
@@ -468,33 +480,68 @@ def grpc_end2end_tests():
             #print(_compatible(fopt, topt), f, t, fopt, topt)
             if not _compatible(fopt, topt):
                 continue
-
             test_short_name = str(t) if not topt.short_name else topt.short_name
-            native.sh_test(
-                name = "%s_test@%s" % (f, test_short_name),
-                data = [":%s_test" % f],
-                srcs = ["end2end_test.sh"],
-                args = [
-                    "$(location %s_test)" % f,
-                    t,
-                ],
-                tags = ["no_linux"] + _platform_support_tags(fopt),
-                flaky = t in fopt.flaky_tests,
-            )
 
+            # Non-Linux platforms
+            for engine in EVENT_ENGINES:
+                if "no_windows" not in engine["tags"] and "no_mac" not in engine["tags"]:
+                    # These platforms do not support multiple polling engines,
+                    # so just create a target for each EventEngine.
+                    native.sh_test(
+                        name = "%s_test@%s@engine=%s" % (f, test_short_name, engine["name"]),
+                        data = [":%s_test" % f],
+                        srcs = ["end2end_test.sh"],
```

Comment:
This script looks like it basically does the same thing as run_with_poller.sh, so I suspect it's not really needed either.

Reply:
replacing `run_with_poller.sh` with cc_tests added roughly 4x the number of cc_tests (and removed that many sh_tests). Doing it here would be around 100x. In the long run I hope to replace sh_test with cc_test, which would allow us to get rid of the intermediate script. That's mostly orthogonal to this work, so I decided to wait on that in case it unearths problems that would block us landing this feature.

While we still have sh_tests, we could replace this script with a simple genrule in the short-term, but it seems kindof trivial.

Reply:
I don't see why the number of cc_test targets matters.  Running the shell here seems like strictly additional overhead without any benefit.

The only real complication I see here is the fact that the cc_test rules are visible in cmake.  But I'd like to see some evidence that that causes a problem before doing anything about it.